objects/oplayercontroller/Step_0.gml

```diff
@@ -114,8 +147,20 @@ if state == eState.CROUCHING {
 	
 	if (movedir == 0 && verticalMoveDir != -1) state = eState.IDLE;
 	if (movedir != 0 && verticalMoveDir != -1) state = eState.WALKING;
-	if verticalMoveDir == 1 state = eState.JUMPSQUAT;
-	
+	if verticalMoveDir == 1 {
+		state = eState.JUMPSQUAT;
+		hsp = walkSpeed * movedir;
+		// Is the player jumping forward?
+		if movedir = image_xscale isJumpingForward = true;

```

Comment:
This condition will always be true, because assignments always return true. Intention appears to be to check equivalence which uses the ```==``` operator.

EDIT: Apparently in some circumstances the assignment operator can function like the equivalence one, but it is widely considered bad form to do so.

Enclose the condition with ```(``` and ```)``` and move the code block into ```{``` and ```}```

Alternatively use a single line condition check replacing both the if and matching else:
```
isJumpingForward = (movedir == image_xscale);